tests/data/vocabulary_test.py

```diff
@@ -119,3 +120,41 @@ def test_set_from_file(self):
         assert vocab.get_token_from_index(4) == "a"
         assert vocab.get_token_from_index(5) == "word"
         assert vocab.get_token_from_index(6) == "another"
+
+    def test_saving_and_loading(self):
+        # pylint: disable=protected-access
+        vocab_dir = os.path.join(self.TEST_DIR, 'vocab_save')
+
+        vocab = Vocabulary(non_padded_namespaces=["a", "c"])
+        vocab.add_token_to_namespace("a0", namespace="a")  # non-padded, should start at 0
+        vocab.add_token_to_namespace("a1", namespace="a")
+        vocab.add_token_to_namespace("a2", namespace="a")
+        vocab.add_token_to_namespace("b2", namespace="b")  # padded, should start at 2
+        vocab.add_token_to_namespace("b3", namespace="b")
+
+        vocab.save_to_files(vocab_dir)             # save to file
+
+        vocab2 = Vocabulary.from_files(vocab_dir,
+                                       non_padded_namespaces=['a', 'c'])
```

Comment:
I think we should figure out a way to save this to disk, too.  It makes the logic for reading all of the files in the directory a little more messy, but I think it's worth it.  You shouldn't have to specify this to load a vocab that we saved ourselves.